content-src/components/SectionMenu/SectionMenu.jsx

```diff
@@ -51,15 +63,19 @@ export class _SectionMenu extends React.PureComponent {
         const { action, id, type, userEvent } = option;
         if (!type && id) {
           option.onClick = () => {
-            props.dispatch(action);
-            if (userEvent) {
-              props.dispatch(
-                ac.UserEvent({
-                  event: userEvent,
-                  source: props.source,
-                })
-              );
+            const addingElements =
+              userEvent === "MENU_ADD_TOPSITE" ||
+              userEvent === "MENU_ADD_SEARCH";
+
+            if (props.collapsed && addingElements) {
+              const {
+                action: expandAction,
+                userEvent: expandEvent,
+              } = SectionMenuOptions.ExpandSection(props);
+              this.dispatchAction(expandAction, expandEvent);
```

Comment:
How about taking out L71 -L75 outside in a separate handleAddWhileCollapsed function. That way we don't need to write a separate dispatchAction

Reply:
Sounds good!